cmd/mt-parrot/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"github.com/grafana/metrictank/cmd/mt-fakemetrics/out"
+	"github.com/grafana/metrictank/cmd/mt-fakemetrics/out/gnet"
+	"github.com/grafana/metrictank/logger"
+	"github.com/grafana/metrictank/schema"
+	"github.com/raintank/met/statsd"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"time"
+)
+
+var (
+	gatewayAddress        string
+	gatewayKey            string
+	orgId                 int
+	partitionCount        int32
+	partitionMethodString string
+	testMetricsInterval   time.Duration
+	queryInterval         time.Duration
+	lookbackPeriod        time.Duration
+	logLevel              string
+	lastPublish           int64
+
+	partitionMethod schema.PartitionByMethod
+	publisher       out.Out
+)
+
+func init() {
+	parrotCmd.Flags().StringVar(&gatewayAddress, "gateway-address", "http://localhost:6059", "the url of the metrics gateway")
+	parrotCmd.Flags().StringVar(&gatewayKey, "gateway-key", "", "the bearer token to include with gateway requests")
+	parrotCmd.Flags().IntVar(&orgId, "org-id", 1, "org id to publish parrot metrics to")
+	parrotCmd.Flags().Int32Var(&partitionCount, "partition-count", 8, "number of kafka partitions in use")
+	parrotCmd.Flags().StringVar(&partitionMethodString, "partition-method", "bySeries", "the partition method in use on the gateway, must be one of bySeries|bySeriesWithTags|bySeriesWithTagsFnv")
+	parrotCmd.Flags().DurationVar(&testMetricsInterval, "test-metrics-interval", 10*time.Second, "interval to send test metrics")
+	parrotCmd.Flags().DurationVar(&queryInterval, "query-interval", 10*time.Second, "interval to query to validate metrics")
```

Comment:
we need to check that these intervals are multiples of `time.Second`

Reply:
23a250ba07a5c7448344f9d8fe7ef4e0dc5cf4d8